capture.js

```diff
@@ -0,0 +1,47 @@
+const capture = {
+
+  init() {
+    this.addListeners();
+  },
+
+  addListeners() {
+    // listen for each HTTP response
+    browser.webRequest.onResponseStarted.addListener(this.collectRequests, {urls: ['<all_urls>']}
+    );
+  },
+
+  /*
+  * makes calls to the storage API for each
+  * first and third party request
+  */
+  async collectRequests(response) {
+    const tab = await browser.tabs.get(response.tabId);
+    let tabUrl = new URL(tab.url);
+    tabUrl = tabUrl.hostname;
+    let targetUrl = new URL(response.url);
+    targetUrl = targetUrl.hostname;
+    let originUrl = new URL(response.originUrl);
+    originUrl = originUrl.hostname;
+
+    //capture first party requests
+    const re = /^about:/;
```

Comment:
`URL(urlString).scheme === "about"` is likely cleaner however I do similar, just wanted to highlight :)